server/issue.go

```diff
@@ -193,6 +193,75 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
 	return http.StatusOK, nil
 }
 
+func (p *Plugin) assignJiraIssue(mmUserId, issueKey, assignee string) (string, error) {
+	ji, err := p.LoadCurrentJIRAInstance()
+	if err != nil {
+		return "", err
+	}
+
+	jiraUser, err := ji.GetPlugin().LoadJIRAUser(ji, mmUserId)
+	if err != nil {
+		return "", err
+	}
+
+	jiraClient, err := ji.GetJIRAClient(jiraUser)
+	if err != nil {
+		return "", err
+	}
+
+	// check for valid issue key
+	_, _, err = jiraClient.Issue.Get(issueKey, nil)
+	if err != nil {
+		errorMsg := fmt.Sprintf("We couldn't find the issue key `%s`.  Please confirm the issue key and try again.", issueKey)
+		return errorMsg, nil
+	}
+
+	// Get list of assignable assignees
+	url := fmt.Sprintf("rest/api/3/user/assignable/search?issueKey=%s&query=%s", issueKey, assignee)
+	req, err := jiraClient.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	res, err := jiraClient.Do(req, nil)
+	if err != nil {
+		if res.Response.StatusCode == 401 {
+			return "You do not have the appropriate permissions to perform this action. Please contact your Jira administrator.", nil
+		}
+		return "", err
+	}
+
+	var data []*jira.User
+	defer res.Body.Close()
```

Comment:
res.Body.Close() must be called even in the case of an error, if `resp` != nil

Reply:
@levb see proposed change shown below.  This code would be placed immediately after the request on line 220.  Is that correct?  We don't have to close the body if resp == nil?
```
	if res != nil {
		defer res.Body.Close()
	}
```

Reply:
I suggest putting it as your first line after checking the error from the request (line 227 above):
```go
if err != nil {
    // do stuff
}
defer resp.Body.Close()
```

Reply:
Alternatively, pass in `&data` to `jiraClient.Do` and it will do all this for you.

Reply:
Nice catch.  I pulled this section of code from hackathon code and didn't realize the optional interface input to `jiraClient.Do` 

That will handle this code for me and reduce the custom code nicely. Thanks @cpoile